### ISSUE TITLE
[GraphQL-Enterprise] Invalid queries returned with ruby error instead of proper graphql error

### ISSUE THREAD

**Issue by davelooi** [rmosolgo/graphql-ruby#3703]
**Describe the bug**

When using graphql-enterprise object cache feature, graphql raise error 500 instead of returning proper error on invalid query.

**Versions**

`graphql` version:

    graphql (1.12.18)
    graphql-client (0.17.0)
      activesupport (>= 3.0)
      graphql (~> 1.10)
    graphql-enterprise (1.0.0)
      graphql (>= 1.12.0)
      graphql-pro (>= 1.19.0)
    graphql-pro (1.19.2)
      graphql (>= 1.7.5)
`rails` (or other framework):

    rails (6.1.4.1)
      actioncable (= 6.1.4.1)
      actionmailbox (= 6.1.4.1)
      actionmailer (= 6.1.4.1)
      actionpack (= 6.1.4.1)
      actiontext (= 6.1.4.1)
      actionview (= 6.1.4.1)
      activejob (= 6.1.4.1)
      activemodel (= 6.1.4.1)
      activerecord (= 6.1.4.1)
      activestorage (= 6.1.4.1)
      activesupport (= 6.1.4.1)
      bundler (>= 1.15.0)
      railties (= 6.1.4.1)
      sprockets-rails (>= 2.0.0)

**GraphQL schema**

```ruby
field :to_string, String, null: false do
  argument :int, Int, required: true
end

def to_string(int:)
  int.to_s
end
```

**GraphQL query**

```graphql
query {
  viewer {
    toString(int: 100000000000)
  }
}
```

**Steps to reproduce**

Execute that query

**Expected behavior**

Return with proper error
```json
{
  "errors": [
    {
      "message": "Argument 'int' on Field 'toString' has an invalid value (100000000000). Expected type 'Int!'.",
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ],
      "path": [
        "query",
        "viewer",
        "toString",
        "int"
      ],
      "extensions": {
        "code": "argumentLiteralsIncompatible",
        "typeName": "Field",
        "argumentName": "int"
      }
    }
  ]
}
```

**Actual behavior**

What specifically went wrong?

Returns with error 500 and ruby backtrace.

<details>
<summary>Click to view exception backtrace</summary>

```json
{
  "error": {
    "message": "undefined method `[]' for nil:NilClass",
    "backtrace": [
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-enterprise-1.0.0/lib/graphql/enterprise/object_cache.rb:218:in `context_fingerprint_for'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-enterprise-1.0.0/lib/graphql/enterprise/object_cache.rb:291:in `cache_fingerprint_for'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-enterprise-1.0.0/lib/graphql/enterprise/object_cache.rb:310:in `prepare'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-enterprise-1.0.0/lib/graphql/enterprise/object_cache.rb:517:in `block in trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-enterprise-1.0.0/lib/graphql/enterprise/object_cache.rb:516:in `each'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-enterprise-1.0.0/lib/graphql/enterprise/object_cache.rb:516:in `trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing.rb:83:in `call_tracers'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing.rb:67:in `trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/analysis/ast.rb:34:in `analyze_multiplex'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/multiplex.rb:207:in `block in instrument_and_analyze'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:29:in `block (2 levels) in apply_instrumenters'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:46:in `block (2 levels) in each_query_call_hooks'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:41:in `each_query_call_hooks'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:45:in `block in each_query_call_hooks'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:72:in `call_hooks'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:44:in `each_query_call_hooks'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:27:in `block in apply_instrumenters'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:72:in `call_hooks'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/instrumentation.rb:26:in `apply_instrumenters'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/multiplex.rb:187:in `instrument_and_analyze'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/multiplex.rb:62:in `block in run_queries'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing.rb:81:in `call_tracers'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing.rb:83:in `block in call_tracers'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing/platform_tracing.rb:26:in `block in trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing/data_dog_tracing.rb:37:in `block in platform_trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/ddtrace-0.53.0/lib/ddtrace/tracer.rb:283:in `trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing/data_dog_tracing.rb:18:in `platform_trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing/platform_tracing.rb:25:in `trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing.rb:83:in `call_tracers'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing.rb:83:in `block in call_tracers'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-enterprise-1.0.0/lib/graphql/enterprise/object_cache.rb:524:in `trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing.rb:83:in `call_tracers'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/tracing.rb:67:in `trace'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/multiplex.rb:60:in `run_queries'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/execution/multiplex.rb:50:in `run_all'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/schema.rb:1701:in `multiplex'",
      "/home/looi/.asdf/installs/ruby/2.7.4/lib/ruby/gems/2.7.0/gems/graphql-1.12.18/lib/graphql/schema.rb:1672:in `execute'",
      ...
    ]
  },
  "data": {}
}
```

</details>

**Additional context**

This happens after adding graphql-enterprise and enabling object cache.

```ruby
use GraphQL::Enterprise::ObjectCache, redis: REDIS_CLIENT, cache_introspection: true
```

**Comment by rmosolgo** [comment]
Oops, sorry about the trouble! Looks like I missed a spot where it should check that the query analyzer finished successfully. I just shipped graphql-enterprise 1.0.1 with a fix for this issue -- please let me know if you run into any more trouble with it!

**Comment by davelooi** [comment]
It worked great. Thank you for your quick response.
